velox/exec/Operator.cpp

```diff
@@ -294,6 +294,29 @@ OperatorStats Operator::stats(bool clear) {
   return ret;
 }
 
+uint32_t Operator::outputBatchRows() const {
+  const auto& queryConfig = operatorCtx_->task()->queryCtx()->queryConfig();
+
+  if (!estimateOutputRowSize().has_value()) {
+    return queryConfig.defaultOutputBatchRows();
+  }
+
+  uint64_t rowSize = estimateOutputRowSize().value();
+
+  if (FOLLY_UNLIKELY(rowSize <= 0)) {
```

Comment:
Should we throw instead? VELOX_CHECK_GT(rowSize, 0)

Reply:
Okay with it. Revised.